photon-ml/src/main/scala/com/linkedin/photon/ml/avro/data/NameAndTerm.scala

```diff
@@ -18,13 +18,21 @@ package com.linkedin.photon.ml.avro.data
  * A compact way to represent the feature key as (name, term) pair.
  * @author xazhang
  */
-//TODO: Change the scope of this class and all functions in the companion object to [[com.linkedin.photon.ml.avro]] after Avro related classes/functons are decoupled from the rest of code
+//TODO: Change the scope of this class and all functions in the companion object to [[com.linkedin.photon.ml.avro]]
+//after Avro related classes/functons are decoupled from the rest of code
 protected[ml] case class NameAndTerm(name: String, term: String) {
 
   override def hashCode: Int = {
     (name + NameAndTerm.DELIMITER + term).hashCode
   }
 
+  override def equals(that: Any): Boolean = {
+    that match {
+      case other: NameAndTerm => hashCode == other.hashCode
```

Comment:
Generally you're always supposed to override these two in pairs. The reason is that ```hashCode``` determines an object's identity (in the place of objectId), and ```equals``` (in most cases) is an identity check. Otherwise you can get into weird situations where e.g. multiple objects are collapsed into one in a set, but they're not "equal" to each other. So it's usually best to keep these two in sync.

This could be an exception though -- e.g., perhaps two NameAndTerm values should be considered distinct, even if their name and term values are the same?